Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -27,7 +27,7 @@ public protocol NavigationService: CLLocationManagerDelegate, RouterDataSource,
     /**
      `RoutingProvider`, used to create route.
```

Comment:
This is a very terse description. Let's expand it explaining what 'nil' means and provide a link for more information if there is a place where it explained better.

Reply:
I've expanded the description a little bit, but documentation in `Router` and `NavigationService` is short because users may implement their own classes and `RoutingProvider` may be used differently, so I decided to keep it short.
I'll be adding a new a example to demonstrate RoutingProvider usage in our examples repo, once this feature lands.